bundles/org.eclipse.ui.ide/src/org/eclipse/ui/internal/ide/ChooseWorkspaceDialog.java

```diff
@@ -474,33 +474,37 @@ protected String getUnexpectedPathHint() {
 			}
 			String normalisedPath = path.normalize().toString();
 			String normalisedPathWithSeperator = normalisedPath + File.separator;
-			if (!isWritable(path)) {
-				return NLS.bind(IDEWorkbenchMessages.ChooseWorkspaceDialog_NotWriteablePathWarning, normalisedPath);
-			}
 			if (normalisedPathWithSeperator.contains(TILDE)) {
 				return NLS.bind(IDEWorkbenchMessages.ChooseWorkspaceDialog_TildeNonExpandedWarning, normalisedPath);
 			}
 			if (!workspaceLocation.equalsIgnoreCase(normalisedPath)
 					&& !workspaceLocation.equalsIgnoreCase(normalisedPathWithSeperator)) {
 				return NLS.bind(IDEWorkbenchMessages.ChooseWorkspaceDialog_ResolvedAbsolutePath, normalisedPath);
 			}
+			if (!mayBeWritable(path)) {
+				return NLS.bind(IDEWorkbenchMessages.ChooseWorkspaceDialog_NotWriteablePathWarning, normalisedPath);
+			}
 		}
 		return ""; //$NON-NLS-1$
 	}
 
-	/**
-	 * @param path
-	 * @return
-	 */
-	private boolean isWritable(Path path) {
-		if (Files.exists(path)) {
-			return Files.isWritable(path);
-		}
-		Path parent = path.getParent();
-		if (parent != null) {
-			return isWritable(parent);
+	private boolean mayBeWritable(Path path) {
```

Comment:
I find the name a little awkward.  English you have may be and maybe and they have subtly different mean, allowed to be versus possibly.  In the end, if this method returns true, the file is definitely writable I think, right?  I think all the bugs are this returning false when it should return true. 

I would just leave the method as isWritable with the description of it being unreliable...

Reply:
depending on how the method is implemented (io/nio) the returned value did show wrong values in both directions. It is unreliable and should reflect that with the name. Changed to "maybe"